curl-config.c.in

```diff
@@ -0,0 +1,429 @@
+/***************************************************************************
+ *                                  _   _ ____  _
+ *  Project                     ___| | | |  _ \| |
+ *                             / __| | | | |_) | |
+ *                            | (__| |_| |  _ <| |___
+ *                             \___|\___/|_| \_\_____|
+ *
+ * Copyright (C) Daniel Stenberg, <daniel@haxx.se>, et al.
+ *
+ * This software is licensed as described in the file COPYING, which
+ * you should have received as part of this distribution. The terms
+ * are also available at https://curl.se/docs/copyright.html.
+ *
+ * You may opt to use, copy, modify, merge, publish, distribute and/or sell
+ * copies of the Software, and permit persons to whom the Software is
+ * furnished to do so, under the terms of the COPYING file.
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTY OF ANY
+ * KIND, either express or implied.
+ *
+ * SPDX-License-Identifier: curl
+ *
+ ***************************************************************************/
+
+#include <ctype.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+/* install prefix */
+/*
+  FIXME: if someone uses --prefix with cmake --install, this will not match
+  the actual installation directory. should run a CMake script on install to
+  reconfigure and recompile as necessary in order to ensure coherency. note
+  tht the pkg-config file generated will have the same problem.
+*/
+static const char *prefix = "@prefix@";  /* CMAKE_INSTALL_PREFIX */
+/* exec prefix (fully expanded). we don't use this however */
+/* static const char *exec_prefix = "@lit_exec_prefix@"; */
+/* pkg-config C/C++ flags */
+static const char *cppflag_curl_staticlib = "@LIBCURL_PC_CFLAGS@";
+/* space-separated feature list */
+static const char *curl_features = "@SUPPORT_FEATURES@";
+/* space-separated protocol list */
+static const char *curl_protocols = "@SUPPORT_PROTOCOLS@";
+/* libcurl version, major.minor.patch[-info] */
+static const char *curl_version = "@CURLVERSION@";
+/* curl-config help text */
+/*
+  TODO: --configure should be used for the CMake configuration variables. not
+  yet sure if we should include only user-supplied cache variables or also any
+  feature variables that may be set during the running of CMake.
+*/
+static const char *usage = "Usage: curl-config [OPTION]\n"
+  "\n"
+  "This version of curl-config was configured with CMake @CMAKE_VERSION@.\n"
+  "\n"
+  "Available values for OPTION include:\n"
+  "\n"
+  "  --built-shared says 'yes' if libcurl was built shared\n"
+  "  --ca                  CA bundle install path\n"
+  "  --cc                  compiler\n"
+  "  --cflags              preprocessor and compiler flags\n"
+  "  --checkfor [version]  check for (lib)curl of the specified version\n"
+  "  --configure           the arguments given to configure when building curl\n"
```

Comment:
This still prints nothing (see `CMakeLists.txt` TODO for `CONFIGURE_OPTIONS`).

It could be changed to some CMake configure options, e.g. `-DCURL_USE_SCHANNEL=ON -DCURL_USE_LIBPSL=OFF` but how exactly this is done probably will require some CMake scripting to get a consistent format.